### ISSUE TITLE
Beneficiary redemption rewards are incorrect for not yet executed proposals.

### ISSUE THREAD

**Issue by dev-matan-tsuberi** [daostack/alchemy#76]
These [three lines](https://github.com/daostack/alchemy/blob/dev/src/actions/arcActions.ts#L387) don't work for proposals which ended and are not executed.
We need to "simulate" what would it return if it were executed.

**Comment by dkent600** [comment]
I don't think this will help with the immediate problem, but just wanted to make sure you are aware of the Arc.js function `ContributionReward.getBeneficiaryRewards`.

**Comment by dev-matan-tsuberi** [comment]
@dkent600 It will still return zero. It's because of [this line](https://github.com/daostack/arc/blob/master/contracts/universalSchemes/ContributionReward.sol#L362) in Arc.

**Comment by dkent600** [comment]
Yes, I just wanted to make sure you were aware of the function

**Comment by tibetsprague** [comment]
Hmm, seems like arc should probably do this though, I'd rather not duplicate the code. Though in the short term we may have to

**Comment by dev-matan-tsuberi** [comment]
@tibetsprague Arc is working correctly, the only thing is to maybe refactor it to be more comfortable for clients, but this will probably be a breaking change. In any case, this inconvenience can be covered by either Arc.js or Alchemy (preferably Arc.js).

**Comment by dkent600** [comment]
I agree with @tibetsprague that the logic for computing the amount of rewards should be kept inside of Arc.

**Comment by tibetsprague** [comment]
however I also agree that it is unlikely we will get arc to change int the short term, though I will create a GitHub issue to start the conversation. in the meantime arc.js computeRemainingReward could use the total number of periods for the proposal instead of getPeriodsToPay when the proposal is not yet executed

**Comment by tibetsprague** [comment]
https://github.com/daostack/arc/issues/468
